tests/gzip.rs

```diff
@@ -48,3 +48,51 @@ fn gzip_stream_large() {
         Vec::from_iter(bytes[0].iter().chain(bytes[1].iter()).cloned())
     );
 }
+
+#[test]
+fn decompressed_gzip_stream_1() {
+    use async_compression::stream::gzip;
+
+    let stream = stream::iter(vec![
+        Bytes::from_static(&[1, 2, 3]),
+        Bytes::from_static(&[4, 5, 6]),
+    ]);
+    let compressed = gzip::GzipStream::new(stream.map(Ok), gzip::Compression::default());
+    let decompressed = gzip::DecompressedGzipStream::new(compressed);
+    let data: Vec<_> = block_on(decompressed.collect());
+    let data: io::Result<Vec<_>> = data.into_iter().collect();
+    let data: Vec<u8> = data.unwrap().into_iter().flatten().collect();
+    assert_eq!(data, vec![1, 2, 3, 4, 5, 6]);
+}
+
+#[test]
+fn decompressed_gzip_stream_2() {
+    use async_compression::stream::gzip;
+
+    let stream = stream::iter(vec![
+        Bytes::from_static(&[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
```

Comment:
For testing the behaviour here I think you'd want to pre-calculate the gzipped version of some data and then slice it in various ways to check that they're all dealt with correctly, e.g. the main ones I can think of:

```rust
let header: [u8; 10] = [...];
let body: [u8; _] = [...];
let trailer: [u8; 8] = [...];

// The entirety in one chunk
let stream = stream::iter(vec![
    Bytes::from(Vec::from_iter(header.iter().chain(body.iter()).chain(trailer.iter()).cloned())),
]);

// Header, body and trailer in separate chunks, similar to how `GzipStream` outputs it.
let stream = stream::iter(vec![
    Bytes::from(&header[..]),
    Bytes::from(&body[..]),
    Bytes::from(&trailer[..]),
]);

// Header, body and trailer each split across multiple chunks, no mixing
let stream = stream::iter(vec![
    Bytes::from(&header[0..5]),
    Bytes::from(&header[5..10]),
    Bytes::from(&body[0..half]),
    Bytes::from(&body[half..end]),
    Bytes::from(&trailer[0..4]),
    Bytes::from(&trailer[4..8]),
]);

// Header, body and trailer split across multiple chunks and mixed together
let stream = stream::iter(vec![
    Bytes::from(&header[0..5]),
    Bytes::from(Vec::from_iter(header[5..10].iter().chain(body[0..half].iter()).cloned())),
    Bytes::from(Vec::from_iter(body[half..end].iter().chain(trailer[0..4].iter()).cloned())),
    Bytes::from(&trailer[4..8]),
]);

Reply:
Oh, I can try this. Good suggestion!